src/main/java/Task.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * The Task class is a parent class for the different types of tasks that the user
+ * input will generate.
+ */
+class Task {
+    private static String horizontalLine = "__________________________________________";
+    protected String taskName;
+    protected static int taskNumber = 1; // for Level 2
+    protected int currentTaskNumber;
+    protected boolean isDone; // for Level 3
+
+    /**
+     * Constructor for a Task takes in a String task name that is specified by the user.
+     *
+     * @param taskName The String input by the user.
+     */
+    Task(String taskName) {
+        this.isDone = false;
+        this.taskName = taskName;
+        currentTaskNumber = taskNumber;
```

Comment:
if we are using a static counter, then in this case, the number of the task in your list will never decrement? And also not sure if you've handled this when loading from storage. 
I guess this is a design decision, but personally I would prefer a task list where number of the tasks decrements as I delete them